### PR TITLE
Remove ETC2/EAC compressed texture formats from WebGL 2.0 core

### LINKED ISSUE #2030: Remove ETC2/EAC compressed texture formats from WebGL 2.0 core
On the public_webgl list it has been noted by developers that decompressing ETC2/EAC compressed texture formats in order to claim conformance on devices which don't support them natively is counter-productive; it increases the VRAM consumption by a factor of six. See https://www.khronos.org/webgl/public-mailing-list/archives/1609/threads.php#00000 .

After discussion, it sounds like the best option is to remove these enums from the core WebGL 2.0 specification, and leave them in the WEBGL_compressed_texture_es3 extension:
https://www.khronos.org/registry/webgl/extensions/WEBGL_compressed_texture_es3/

So, for concrete steps:

1) Chrome should implement this draft extension, so there are two implementations (Firefox being the other): http://crbug.com/455568
2) It should be promoted to community approved.
3) The enums should be removed from the WebGL 2.0 spec, and the conformance tests updated to only run these tests if the extension is available.

### PR DESCRIPTION
**Pull request by kainino0x**
This fixes #2030.

- [x] #2047
- [x] #2083

### CODE REVIEW

**Comment by kainino0x** [comment]
Lint is failing because I touched a file which was already failing the lint (it has a weird mix of tabs and spaces).

File: sdk/tests/conformance/extensions/webgl-compressed-texture-s3tc-srgb.html [hunk 1]
```diff
@@ -200,12 +196,8 @@
     }
 
     supportedFormats = gl.getParameter(gl.COMPRESSED_TEXTURE_FORMATS);
-    // There should be exactly 4 formats for WebGL 1.0 and 14 formats for WebGL 2.0.
-    if (contextVersion < 2) {
-        shouldBe("supportedFormats.length", "4");
-    } else {
-        shouldBe("supportedFormats.length", "14");
-    }
+    // There should be exactly 4 formats for both WebGL 1.0 WebGL 2.0.
```
Comment (kenrussell) [inline sdk/tests/conformance/extensions/webgl-compressed-texture-s3tc-srgb.html]: typo: "WebGL 1.0 and WebGL 2.0"

**Comment by kenrussell** [comment]
Uploaded #2056 to address the lint problems. Please rebase once it lands. Thanks.

**Comment by kainino0x** [comment]
I added the WebGL 2 spec changes to this PR. PTAL when you have a chance @kenrussell, thanks.

Thanks @zmo for help on the spec editing.

**Comment by zhenyao** [comment]
LGTM

File: specs/latest/2.0/index.html [hunk 2]
```diff
@@ -3700,6 +3691,26 @@ <h2 class="no-toc">Table of contents</h2>
...
     </div>
 
+    <h3><a name="NO_ETC2_EAC">No ETC2 and EAC compressed texture formats</a></h3>
+    <p>OpenGL ES 3.0 requires support for the following ETC2 and EAC compressed texture formats:
+       COMPRESSED_R11_EAC, COMPRESSED_SIGNED_R11_EAC, COMPRESSED_RG11_EAC,
+       COMPRESSED_SIGNED_RG11_EAC, COMPRESSED_RGB8_ETC2, COMPRESSED_SRGB8_ETC2,
+       COMPRESSED_RGB8_PUNCHTHROUGH_ALPHA1_ETC2, COMPRESSED_SRGB8_PUNCHTHROUGH_ALPHA1_ETC2,
+       COMPRESSED_RGBA8_ETC2_EAC, and COMPRESSED_SRGB8_ALPHA8_ETC2_EAC.
+
+    <p>These texture formats are not supported in WebGL 2.0.
```
Comment (kenrussell) [inline specs/latest/2.0/index.html]: I'd probably have said "not supported by default".

**Pull request merged**